src/cryptonote_core/cryptonote_tx_utils.h

```diff
@@ -30,47 +30,15 @@
 
 #pragma once
 #include "cryptonote_basic/cryptonote_format_utils.h"
-#include <boost/serialization/vector.hpp>
-#include <boost/serialization/utility.hpp>
 
 namespace cryptonote
 {
   //---------------------------------------------------------------
   bool construct_miner_tx(size_t height, size_t median_size, uint64_t already_generated_coins, size_t current_block_size, uint64_t fee, const account_public_address &miner_address, transaction& tx, const blobdata& extra_nonce = blobdata(), size_t max_outs = 999, uint8_t hard_fork_version = 1);
 
-  struct tx_source_entry
-  {
-    typedef std::pair<uint64_t, rct::ctkey> output_entry;
-
-    std::vector<output_entry> outputs;  //index + key + optional ringct commitment
-    size_t real_output;                 //index in outputs vector of real output_entry
-    crypto::public_key real_out_tx_key; //incoming real tx public key
-    size_t real_output_in_tx_index;     //index in transaction outputs vector
-    uint64_t amount;                    //money
-    bool rct;                           //true if the output is rct
-    rct::key mask;                      //ringct amount mask
-
-    void push_output(uint64_t idx, const crypto::public_key &k, uint64_t amount) { outputs.push_back(std::make_pair(idx, rct::ctkey({rct::pk2rct(k), rct::zeroCommit(amount)}))); }
-  };
-
-  struct tx_destination_entry
-  {
-    uint64_t amount;                    //money
-    account_public_address addr;        //destination address
-
-    tx_destination_entry() : amount(0), addr(AUTO_VAL_INIT(addr)) { }
-    tx_destination_entry(uint64_t a, const account_public_address &ad) : amount(a), addr(ad) { }
-
-    BEGIN_SERIALIZE_OBJECT()
-      VARINT_FIELD(amount)
-      FIELD(addr)
-    END_SERIALIZE()
-  };
-
   //---------------------------------------------------------------
-  crypto::public_key get_destination_view_key_pub(const std::vector<tx_destination_entry> &destinations, const account_keys &sender_keys);
```

Comment:
This function along with the above two structs are moved into cryptonote_format_utils. Strictly speaking this is not really necessary, but I did so because I wanted to keep cryptonote_tx_utils (which was introduced by my previous PR) to the absolute minimum.